chips/stm32f4xx/src/lib.rs

```diff
@@ -9,6 +9,7 @@
 #![crate_name = "stm32f4xx"]
 #![crate_type = "rlib"]
 #![no_std]
+#![feature(core_panic)]
```

Comment:
Nightly feature.

Reply:
This was added because of the `check_and_panic!` macro:

```rust
macro_rules! check_and_panic {
	($left:expr, $right:expr, $clocks: ident) => {
		match (&$left, &$right) {
			(left_val, right_val) => {
				if !(*left_val == *right_val) {
					set_default_configuration($clocks);
					let kind = core::panicking::AssertKind::Eq;
					// The reborrows below are intentional. Without them, the stack slot for the
					// borrow is initialized even before the values are compared, leading to a
					// noticeable slow down.
					core::panicking::assert_failed(
						kind,
						&*left_val,
						&*right_val,
						core::option::Option::None,
					);
				}
			}
		};
	};
}
```

This macro ensures that if a test fails, the system clock is reverted back to
its default configuration. This prevents messing up with the UART baud rate.

A solution to this problem could be:

```rust
macro_rules! check_and_panic {
	($left:expr, $right:expr, $clocks: ident) => {
		match (&$left, &$right) {
			(left_val, right_val) => {
				if *left_val != *right_val {
					set_default_configuration($clocks);
				}
				assert_eq!($left, $right);
			}
		};
	};
}
```

The downside of this approach is that `left` and `right` are evaluated twice.

Another solution is to use `panic!` with a custom message for each test. The
downside of this approach is evident: one has to write a custom panic message
for each occurrence of `check_and_panic`.

Do you have a better solution?

Reply:
Implemented the first solution.